packages/sanity/src/core/form/studio/inputResolver/fieldResolver.tsx

```diff
@@ -6,35 +6,45 @@ import {ReferenceField} from '../../inputs/ReferenceInput/ReferenceField'
 import {FieldMember} from '../../store'
 import {FormField, FormFieldSet} from '../../components'
 import {ChangeIndicator} from '../../../changeIndicators'
+import {useField} from '../../field'
 import {getTypeChain} from './helpers'
 
 function BooleanField(field: FieldProps) {
+  const {onMouseEnter, onMouseLeave} = useField()
+
   return (
     <ChangeIndicator
       path={field.path}
       hasFocus={Boolean(field.inputProps.focused)}
       isChanged={field.inputProps.changed}
+      onMouseEnter={onMouseEnter}
+      onMouseLeave={onMouseLeave}
     >
       {field.children}
     </ChangeIndicator>
   )
 }
 
 function PrimitiveField(field: FieldProps) {
+  const {onMouseEnter, onMouseLeave} = useField()
+
   return (
     <FormField
+      __unstable_headerActions={field.actions}
+      __unstable_presence={field.presence}
       data-testid={`field-${field.inputId}`}
+      description={field.description}
       inputId={field.inputId}
       level={field.level}
+      onMouseEnter={onMouseEnter}
+      onMouseLeave={onMouseLeave}
       title={field.title}
-      description={field.description}
       validation={field.validation}
-      __unstable_presence={field.presence}
     >
       <ChangeIndicator
-        path={field.path}
         hasFocus={Boolean(field.inputProps.focused)}
         isChanged={field.inputProps.changed}
+        path={field.path}
```

Comment:
meta: would be great to have non-significant changes like this in a separate PR

Reply:
If it's non-significant, then that's a mistake.

Reply:
Ah, I see it moved. Will move it back.